### ISSUE TITLE
Improve usability of toggle + label elements

### ISSUE THREAD

**Issue by quinnkeast** [sourcegraph/sourcegraph#17133]
### Problem

We use the toggle element in a few places in Sourcegraph, sometimes with and sometimes without a label.

With label:

<img width="572" alt="image" src="https://user-images.githubusercontent.com/6304497/104012090-80628900-51af-11eb-8025-918673051355.png">

Without label: 

<img width="858" alt="image" src="https://user-images.githubusercontent.com/6304497/104012107-88bac400-51af-11eb-8b58-321982001113.png">

When we use the toggle with a label, only the toggle is interactive. Clicking the label doesn't cause the toggle to flip. This feels contrary to best practices for usability and accessibility, particularly in relation to checkbox and radio elements, in which the checkbox or radio should be contained within the label, and clicking the label toggles the element. This improves usability and accessibility by massively increasing the interactive area.

### Proposal

We should expand the toggle component to include an optional label, and optional sub-label, as part of the toggle. Then, that label can be part of the triggering element.

What's clickable today:

<img width="466" alt="image" src="https://user-images.githubusercontent.com/6304497/104012422-10083780-51b0-11eb-84b0-5c3872d778c1.png">

What should be clickable:

<img width="466" alt="image" src="https://user-images.githubusercontent.com/6304497/104012446-19919f80-51b0-11eb-9010-c458d1dd31bd.png">

Additionally, we may want to change the label itself depending on the toggle state. In the above example, the label should change from:

>**Active**
>Code monitor will watch for the trigger and run actions in response

to:

>**Inactive**
>Code monitor is not watching for the trigger

**Comment by pjlast** [comment]
@quinnkeast is this issue still relevant? You mention changing label state based on toggle state, but it seems this is already how it functions.

I have an implementation that should work, just want to confirm this is still desired before starting the PR process.

**Comment by pjlast** [comment]
I also imagine this actually consisting of two parts:

1. Changing the Toggle component to allow for a label that forms part of the component and ensuring compatibility with current implementations where the Toggle is used.
2. Converting existing implementations to use the modified Toggle so that the label can be clicked as well.

**Comment by umpox** [comment]
Added this to the [WCAG tracking issue](https://github.com/sourcegraph/sourcegraph/issues/31476) to audit against our accessibility spec and see if we've updated this on any of our pages!

**Comment by felixfbecker** [comment]
Whether clicking the label toggles the toggle or not is a UX question, but not a WCAG violation afaict.
Note that in both cases described in the issue, there is a _label_. One is just located right next to it, the other precedes it further left in the box. But the toggle is always _labelled_ by a descriptive text label.
The only thing that matters for WCAG here is whether the label is associated with the toggle through `aria-labelledby`, for [1.3.1 Info and Relationships](https://www.w3.org/WAI/WCAG21/Understanding/info-and-relationships).
From a check in Code Monitoring that does not seem to be the case.
